tests/lean/refine_error.lean.expected.out

```diff
@@ -1,5 +1,5 @@
-refine_error.lean:5:1: error: type mismatch, expression
-  @pax ℤ int.has_zero
+refine_error.lean:5:1: error: type mismatch, term
+  pax
```

Comment:
Note the small regression -- I don't think it makes sense to use the `until_different` formatter for the types on the term as well in general. After #1674, we may want to just show the original input (if available) instead.